src/sinks/pulsar.rs

```diff
@@ -229,6 +255,14 @@ impl Sink<Event> for PulsarSink {
             "Expected `poll_ready` to be called first."
         );
 
+        let event_time = match &item {
+            Event::Log(log) => log
+                .get(log_schema().timestamp_key())
+                .map(|v| v.as_timestamp().map(|dt| dt.timestamp_millis()))
+                .unwrap_or(None),
+            _ => None,
+        };
```

Comment:
There is a `maybe_as_log` interface that can help here:
```suggestion
        let event_time = item
            .maybe_as_log()
            .then(|log| log.get(log_schema().timestamp_key()))
            .then(|v| v.as_timestamp())
            .map(|dt| dt.timestamp_millis());
```

Reply:
I only found the interface `maybe_as_log_mut`, is that what you mean? I updated the code using the interface, please take a look again.

Reply:
I'm sorry, I thought there was a non-mut `fn maybe_as_log`. I think it would be better to add that simple accessor to the core type than to make all of this `mut` here.